RecoCTPPS/TotemRPLocal/src/CTPPSDiamondRecHitProducerAlgorithm.cc

```diff
@@ -0,0 +1,48 @@
+/****************************************************************************
+*
+* This is a part of CTPPS offline software.
+* Authors:
+*   Laurent Forthomme (laurent.forthomme@cern.ch)
+*
+****************************************************************************/
+
+#include "RecoCTPPS/TotemRPLocal/interface/CTPPSDiamondRecHitProducerAlgorithm.h"
+
+//----------------------------------------------------------------------------------------------------
+
+CTPPSDiamondRecHitProducerAlgorithm::CTPPSDiamondRecHitProducerAlgorithm( const edm::ParameterSet& iConfig ) :
+  ts_to_ns_( iConfig.getParameter<double>( "timeSliceNs" ) ),
+  t_shift_( iConfig.getParameter<int>( "timeShift" ) )
+{}
+
+void
+CTPPSDiamondRecHitProducerAlgorithm::build( const TotemRPGeometry* geom, const edm::DetSetVector<CTPPSDiamondDigi>& input, edm::DetSetVector<CTPPSDiamondRecHit>& output )
+{
+  for ( edm::DetSetVector<CTPPSDiamondDigi>::const_iterator vec = input.begin(); vec != input.end(); ++vec )
+  {
+    const CTPPSDiamondDetId detid( vec->detId() );
+
+    const DetGeomDesc* det = geom->GetDetector( detid );
+    const float x_pos = det->translation().x(),
+                x_width = 2.0 * det->params().at( 0 ), // parameters stand for half the size
+                y_pos = det->translation().y(),
+                y_width = 2.0 * det->params().at( 1 );
```

Comment:
```diff
-                x_width = det->params().at( 0 ),
+                x_width = 2.0 * det->params().at( 0 ), // parameters stand for half the size
                 y_pos = det->translation().y(),
-                y_width = det->params().at( 1 );
+                y_width = 2.0 * det->params().at( 1 );

```
from https://github.com/cms-sw/cmssw/commit/625cee22997613b589698ce15c82b2c535b75237
appears to be the only non-trivial change in reco-related code after the previously signed version
https://github.com/cms-sw/cmssw/pull/17317  30d057a